backend/src/apiserver/client_manager/client_manager.go

```
@@ -531,9 +613,12 @@ func initLogArchive() (logArchive archive.LogArchiveInterface) {
 }
 
 // NewClientManager creates and Init a new instance of ClientManager.
-func NewClientManager() ClientManager {
+func NewClientManager(options *Options) ClientManager {
 	clientManager := ClientManager{}
-	clientManager.init()
+	err := clientManager.init(options)
+	if err != nil {
+		// TODO: treat error
```

I think `NewClientManager` should just return the error.